azure/services/virtualmachines/virtualmachines.go

```diff
@@ -111,6 +114,34 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 		s.Scope.SetAddresses(addresses)
 		s.Scope.SetVMState(infraVM.State)
+
+		spec, ok := vmSpec.(*VMSpec)
+		if !ok {
+			return errors.Errorf("%T is not a valid VM spec", vmSpec)
+		}
+		idsClient := identities.NewClient(s.Scope)
```

Comment:
here instead of creating a client on the fly, let's make it part of the Service, see how we do it above for `publicIPsGetter`